### PR TITLE
[gdal] Update to 3.4.2

### PR DESCRIPTION
**Pull request by dg0yt**
- #### What does your PR fix?  
  Updates GDAL.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all, yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  ~I am still working on this PR.~ yes

### CODE REVIEW

**Comment by JonLiu1993** [comment]
@dg0yt, thanks for the PR, maybe you forgot to change the vcpkg.json file?

**Comment by maz-1** [comment]
I think you need to escape some dollar signs when fixing pkgconfig.
from
`vcpkg_replace_string("${pc_file_debug}" "${prefix}/../../include" "${prefix}/../include")`
to
`vcpkg_replace_string("${pc_file_debug}" "\${prefix}/../../include" "\${prefix}/../include")`

**Comment by dg0yt** [comment]
> I think you need to escape some dollar signs when fixing pkgconfig. from `vcpkg_replace_string("${pc_file_debug}" "${prefix}/../../include" "${prefix}/../include")` to `vcpkg_replace_string("${pc_file_debug}" "\${prefix}/../../include" "\${prefix}/../include")`

Thanks, confirmed. Affects `CONFIG_INST_CFLAGS`, `Cflags`.

**Comment by maz-1** [comment]
I treid to fix pkgconfig support on windows, coudl you take a look?
https://gist.github.com/maz-1/42a776308f65010eb60e89eb4e777728

**Comment by dg0yt** [comment]
> I treid to fix pkgconfig support on windows, coudl you take a look? https://gist.github.com/maz-1/42a776308f65010eb60e89eb4e777728

a) I don't want to add pkgconfig support for windows (`nmake`) at this point. I'm working here (#22392) and with upstream to switch to the CMake build system with GDAL 3.5. This is from [GDAL CI](https://github.com/OSGeo/gdal/runs/5562763771?check_suite_focus=true):
~~~
CONFIG_VERSION=3.5.0dev
CONFIG_INST_PREFIX=D:/a/gdal/gdal/install-gdal
CONFIG_INST_LIBS= -LIBPATH:D:/a/gdal/gdal/install-gdal/lib gdal
CONFIG_INST_CFLAGS= -ID:/a/gdal/gdal/install-gdal/include
CONFIG_INST_DATA= D:/a/gdal/gdal/install-gdal/share/gdal
prefix=${CONFIG_INST_PREFIX}
exec_prefix=${prefix}
libdir=${exec_prefix}/lib
includedir=${exec_prefix}/include
datadir=${CONFIG_INST_DATA}

Name: libgdal
Description: Geospatial Data Abstraction Library
Version: ${CONFIG_VERSION}
Libs: ${CONFIG_INST_LIBS}
Libs.private:  C:/Miniconda/envs/gdalenv/Library/lib/iconv.lib C:/Miniconda/envs/gdalenv/Library/lib/charset.lib C:/Miniconda/envs/gdalenv/Library/lib/libssl.lib C:/Miniconda/envs/gdalenv/Library/lib/libcrypto.lib crypt32 C:/Miniconda/envs/gdalenv/Library/lib/liblzma.lib C:/Miniconda/envs/gdalenv/Library/lib/libdeflate.lib C:/Miniconda/envs/gdalenv/Library/lib/liblz4.lib C:/Miniconda/envs/gdalenv/Library/lib/blosc.lib wbemuuid C:/Miniconda/envs/gdalenv/Library/lib/armadillo.lib C:/Miniconda/envs/gdalenv/Library/lib/mkl_intel_lp64_dll.lib "C:/Program Files/LLVM/lib/libiomp5md.lib" C:/Miniconda/envs/gdalenv/Library/lib/mkl_intel_thread_dll.lib C:/Miniconda/envs/gdalenv/Library/lib/mkl_core_dll.lib C:/Miniconda/envs/gdalenv/Library/lib/qhull_r.lib C:/Miniconda/envs/gdalenv/Library/lib/geotiff_i.lib C:/Miniconda/envs/gdalenv/Library/lib/tiff.lib ws2_32 C:/Miniconda/envs/gdalenv/Library/lib/jpeg.lib C:/Miniconda/envs/gdalenv/Library/lib/libpng.lib C:/Miniconda/envs/gdalenv/Library/lib/Lerc.lib C:/Miniconda/envs/gdalenv/Library/lib/zstd.lib C:/Miniconda/envs/gdalenv/Library/lib/z.lib C:/Miniconda/envs/gdalenv/Library/lib/pcre2-8.lib C:/Miniconda/envs/gdalenv/Library/lib/spatialite_i.lib C:/Miniconda/envs/gdalenv/Library/lib/libexpat.lib C:/Miniconda/envs/gdalenv/Library/lib/sqlite3.lib C:/Miniconda/envs/gdalenv/Library/lib/libcurl.lib C:/Miniconda/envs/gdalenv/Library/lib/xerces-c_3.lib "C:/Program Files (x86)/Windows Kits/10/Lib/10.0.22000.0/um/x64/odbccp32.lib" "C:/Program Files (x86)/Windows Kits/10/Lib/10.0.22000.0/um/x64/WS2_32.Lib" legacy_stdio_definitions.lib odbccp32.lib C:/Miniconda/envs/gdalenv/Library/lib/xml2.lib C:/Miniconda/envs/gdalenv/Library/lib/geos_c.lib psapi C:/Miniconda/envs/gdalenv/Library/lib/proj.lib C:/Miniconda/envs/gdalenv/Library/lib/json-c.lib
Cflags: ${CONFIG_INST_CFLAGS}
~~~

b) You can make suggestion as PR against my repo and branch. That's much easier to review and apply, and it even carries proper authorship.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
All features are tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

**Review by BillyONeal — APPROVED** [review]

File: ports/gdal/vcpkg.json [hunk 1]
```diff
@@ -5,6 +5,10 @@
   "homepage": "https://gdal.org",
   "license": null,
   "dependencies": [
+    {
+      "name": "atlmfc",
+      "platform": "windows & !uwp"
```
Comment (dg0yt) [inline ports/gdal/vcpkg.json:10]: Now mingw is broken, isn't it?

Comment (dg0yt) [reply ports/gdal/vcpkg.json:10]: Okay.
- `atlmfc` just checks for `afxres.h` at build time.
- MSYS2 mingw installs the header. `atlmfc` builds, and so there is no negative effect on `gdal:*mingw*`.
- `arm64-windows` seems to have the ATL headers, too, but doesn't provide the corresponding `atls.lib`.  
   (We couldn't build the previous releases of GDAL for this reason: 
   https://dev.azure.com/vcpkg/public/_build/results?buildId=68896&view=results)
- Testing for `afxres.h` at  `atlmfc` build time will be of limited value when artifacts are installed on another machine from a shared binary cache.

**Pull request merged**